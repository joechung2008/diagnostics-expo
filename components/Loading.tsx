import { useTheme } from '@/components/ThemeContext';
import React from 'react';
import { ActivityIndicator, StyleSheet, Text, View } from 'react-native';

interface LoadingProps {
  message: string;
}

const Loading = ({ message }: LoadingProps) => {
  const { colors } = useTheme();

  const styles = StyleSheet.create({
    loading: {
      alignItems: 'center',
      flex: 1,
      justifyContent: 'center',
    },
    message: {
      color: colors.text,
      fontSize: 14,
      marginTop: 12,
    },
  });

  return (
    <View style={styles.loading}>
      <ActivityIndicator
        accessibilityLabel={message}
        size="large"
        color={colors.primary}
      />
      {message ? <Text style={styles.message}>{message}</Text> : null}
    </View>
  );
};

export default Loading;
