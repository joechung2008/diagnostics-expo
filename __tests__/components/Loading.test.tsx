import Loading from '@/components/Loading';
import { render } from '@testing-library/react-native';
import React from 'react';

jest.mock('@/components/ThemeContext', () => ({
  useTheme: () => ({
    colors: {
      background: '#ffffff',
      surface: '#f5f5f5',
      text: '#000000',
      primary: '#00ff00',
    },
  }),
}));

describe('Loading component', () => {
  it('renders loading indicator and message', () => {
    const { toJSON } = render(<Loading message="Loading extension..." />);
    expect(toJSON()).toMatchSnapshot();
  });
});
