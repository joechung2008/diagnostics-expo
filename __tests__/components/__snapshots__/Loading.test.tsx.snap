// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component renders loading indicator and message 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    accessibilityLabel="Loading extension..."
    color="#00ff00"
    size="large"
  />
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 14,
        "marginTop": 12,
      }
    }
  >
    Loading extension...
  </Text>
</View>
`;
